src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpers.cs

```diff
@@ -0,0 +1,521 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
```

Comment:
nit: Given the category, should this analyzer be in `Performance` directory and namespace? (this applies to the test class as well)

Reply:
We have a bunch of performance-related analyzers in the Runtime directory; I'm not sure what the rhyme or reason is for the folder they live in.

But regardless, I was going back and forth on whether to have this be Performance or Maintainability, and I decided to switch them to the latter.  They have implications on both, but the high-order bit is helping to make the code cleaner / more maintainable, and any perf improvements that result are bonus.